### ISSUE TITLE
[BUG] - Date / Calender components not updating correctly with locale

### ISSUE THREAD

**Issue by bebbs-dmt** [nextui-org/nextui#3142]
### NextUI Version

2.4.0

### Describe the bug

When updating the locale to use an international calendar, the date and calendar components don't update completely, if at all. This may be related to Next.js, because the CodeSandbox environment for [this example](https://nextui.org/docs/components/date-picker#international-calendar) seems to work fine, and that uses standard React.

The calendar seems to partially update, this is observable in the stackblitz link below.

### Your Example Website or App

https://stackblitz.com/edit/stackblitz-starters-caucmz?file=app%2Fpage.tsx

### Steps to Reproduce the Bug or Issue

1. Open the stackblitz link and wait for the page to compile and load.
2. Observe that the date picker has not properly updated to use the international calendar.

### Expected behavior

Date and calendar components should be updating completely to use the specified locale.

### Operating System Version

Windows 11, developing in WSL Ubuntu 22.04 LTS

### Browser

Chrome

**Comment by wingkwong** [comment]
Able to reproduce in next / vite app.

**Comment by juank1809** [comment]
Can I work on this issue? :)

**Comment by wingkwong** [comment]
@juank1809 assigned. feel free to contact us in Discord if you need any help.

**Comment by wingkwong** [comment]
@juank1809 any updates? if not, I'd like to take over it since I'm working on similar issue.

**Comment by juank1809** [comment]
Hi! I have been trying to resolve this issue for weeks but could not accomplish it. I'm sorry. Feel free to take over :) @wingkwong

**Comment by MohammadHarish52** [comment]
`'use client';

import { DatePicker } from '@nextui-org/date-picker';
import { DateValue, parseAbsoluteToLocal } from '@internationalized/date';
import { I18nProvider } from '@react-aria/i18n';
import { useState } from 'react';

export default function App() {
  const currentDate = new Date(); // Current date and time
  let [date, setDate] = useState<DateValue>(
    parseAbsoluteToLocal(currentDate.toISOString())
  );

  return (
    <div className="flex flex-col gap-4">
      <I18nProvider locale="hi-IN-u-ca-indian">
        <DatePicker
          showMonthAndYearPickers
          variant="flat"
          label="Date"
          value={date}
          onChange={setDate}
        />
      </I18nProvider>
    </div>
  );
}`

used the Date object and it's working fine as of now idk what was the issue about since the string itself was passed how can it show the current time?????

**Comment by sohan01fw** [comment]
is this issue already solved?

**Comment by wingkwong** [comment]
@sohan01fw if it is still open and no linked PR, then no

**Comment by yarikpetrenko** [comment]
Having the same issue, I18nProvider doesn't change anything for me.

**Comment by wingkwong** [comment]
@bebbs-dmt and for those who are having this issue, please check the version of `@react-aria/i18n` that you are using. NextUI is using a fixed version `3.11.1`. If you are using other version, it may cause this issue. From the OP example, you may simply change as follows in your `package.json`.

```diff
-"@react-aria/i18n": "^3.11.1",
+"@react-aria/i18n": "3.11.1",
```

Here's the result. You may need to remove node_modules before reinstalling. I'll update the docs accordingly. 

![image](https://github.com/user-attachments/assets/3184aa90-5c94-4de7-8c6e-94b574517a51)
